tests/acceptance/features/webUISharingPublic/shareByPublicLink.feature

```diff
@@ -653,6 +672,69 @@ Feature: Share by public link
       | lorem.txt       |
       | simple-folder   |
 
+  Scenario: user cannot set an expiry date when creating a public link to a date that is past the enforced max expiry date
+    Given the setting "shareapi_default_expire_date" of app "core" has been set to "yes"
+    And the setting "shareapi_enforce_expire_date" of app "core" has been set to "yes"
+    And user "user1" has logged in using the webUI
+    When the user tries to create a new public link for resource "simple-folder" using the webUI with
+      | expireDate | +8 |
+    Then the user should see an error message on the public link share dialog saying "Cannot set expiration date more than 7 days in the future"
+    And the user "user1" should not have created any shares
+
+  Scenario: user cannot change the expiry date of an existing public link to a date that is past the enforced max expiry date
+    Given the setting "shareapi_default_expire_date" of app "core" has been set to "yes"
+    And the setting "shareapi_enforce_expire_date" of app "core" has been set to "yes"
+    And user "user1" has created a public link with following settings
+      | path       | lorem.txt   |
+      | name       | Public link |
+      | expireDate | +6          |
+    And user "user1" has logged in using the webUI
+    When the user edits the public link named "Public link" of file "lorem.txt" changing following
+      | expireDate | +8 |
+    Then the user should see an error message on the public link share dialog saying "Cannot set expiration date more than 7 days in the future"
+    And user "user1" should have a share with these details:
+      | field       | value       |
+      | share_type  | public_link |
+      | uid_owner   | user1       |
+      | permissions | read        |
+      | path        | /lorem.txt  |
+      | name        | Public link |
+      | expiration  | +6          |
+
+  Scenario: user can set an expiry date when creating a public link to a date that is before the enforced max expiry date
+    Given the setting "shareapi_default_expire_date" of app "core" has been set to "yes"
+    And the setting "shareapi_enforce_expire_date" of app "core" has been set to "yes"
+    And user "user1" has logged in using the webUI
+    When the user creates a new public link for resource "lorem.txt" using the webUI with
+      | expireDate | +7 |
+    Then user "user1" should have a share with these details:
+      | field       | value       |
+      | share_type  | public_link |
+      | uid_owner   | user1       |
+      | permissions | read        |
+      | path        | /lorem.txt  |
+      | name        | Public link |
+      | expiration  | +7          |
+
+  Scenario: user can change the expiry date of an existing public link to a date that is before the enforced max expiry date
```

Comment:
I think this scenario is being duplicated. Exactly similar scenario in L14

Reply:
the L14 scenario was only there for debugging, there have been issues running it because of the "locale" of the month name